crates/gdtk-ast/src/visitor.rs

```diff
@@ -0,0 +1,744 @@
+use crate::ast;
+
+type Range = std::ops::Range<usize>;
+
+pub trait Visitor<'a>: Sized {
```

Comment:
need to verify that this bound doesn't prevent from making `Box<dyn Visitor>`s, which i may need later for user-made lints